packages/shared/src/components/GrowthBookProvider.tsx

```diff
@@ -0,0 +1,127 @@
+import React, {
+  ReactElement,
+  ReactNode,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import { useMutation } from 'react-query';
+import {
+  Context,
+  GrowthBook,
+  GrowthBookProvider as Provider,
+} from '../lib/growthbook';
+import { isProduction } from '../lib/constants';
+import { BootApp, BootCacheData } from '../lib/boot';
+import { decrypt } from './crypto';
+import { apiUrl } from '../lib/config';
+
+export type GrowthBookProviderProps = {
+  app: BootApp;
+  user: BootCacheData['user'];
+  deviceId: string;
+  version?: string;
+  experimentation?: {
+    f: string;
+    e: string[];
+  };
+  fetchMethod?: typeof fetch;
+  children?: ReactNode;
+};
+
+export const GrowthBookProvider = ({
+  app,
+  user,
+  deviceId,
+  version,
+  experimentation,
+  fetchMethod = fetch,
+  children,
+}: GrowthBookProviderProps): ReactElement => {
+  const { mutateAsync: sendAllocation } = useMutation(
+    async (data: { experimentId: string; variationId: string }) => {
+      const res = await fetchMethod(`${apiUrl}/e/x`, {
+        method: 'POST',
+        body: JSON.stringify({
+          event_timestamp: new Date(),
+          user_id: user.id,
+          device_id: deviceId,
+          experiment_id: data.experimentId,
+          variation_id: data.variationId,
+        }),
+        credentials: 'include',
+        headers: {
+          'content-type': 'application/json',
+        },
+      });
+      await res?.text();
+    },
+    {
+      retry: 3,
+    },
+  );
+
+  const callback = useRef<Context['trackingCallback']>();
+  const [gb] = useState<GrowthBook>(
```

Comment:
Could this not be initiated outside of the provider since we don't set it anywhere?

Reply:
I do use the tracingCallback. I need to have the ref available